src/extension.ts

```diff
@@ -50,6 +51,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
             const connectionSucceeded = await connectionPromise;
             if (!connectionSucceeded) {
                 throw new Error(`Connection for ${JSON.stringify(connectionInfo)} failed`);
+            } else if (saveConnection) {
+                let profile: IConnectionProfile = <IConnectionProfile>connectionInfo;
+                await controller.connectionManager.connectionStore.saveProfile(profile);
```

Comment:
Where is the profile normally saved now? Is that just done in the connection picker flow or something? We should make sure there isn't somewhere later on that we can leverage the logic for to save the connection (say by converting this into a profile and setting saveConnection to true here and the passing it on to connect)

Reply:
connection will only get saved when saveconnection is set to true. I'm not sure what you mean here

Reply:
I'm asking how the extension saves profiles currently. Because if I go through the connection profile picker then those connections are saved already. So where is that done? We shouldn't be duplicating logic if there's somewhere else that it's already saving connections we can utilize instead of calling save directly here.

Reply:
sure I can look it up again but what happens is, if connect using the profile picker, then connect method is called and then save method. it always saves after. but in the API we don't want to save all the connections using the API so that's why we give an option to call save if the option is set to true.

Reply:
Can you link where that's done?

Also what about my earlier question regarding calling connect for duplicate profiles?

Reply:
I made the change in the code to do the same as connection is saved from picker.